### DISCUSSION TITLE
Would it be possible to separate widgets and styles into separate files with CustomTkinter?

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [TomSchimansky/CustomTkinter#2218] [Q&A] [answered]
Something similar to html and css

**stylesheet.py**:
```
import customtkinter

large_widget_sty = (row=0, column=0, padx=20, pady=20, sticky="ew", columnspan=2)
same_row_sty = (row=1, column=0, padx=20, pady=(0, 20), sticky="w")
```

**app.py**:
```
from stylesheet import *

class App(customtkinter.CTk):
    def __init__(self):
        super().__init__()

        self.title("my app")
        self.geometry("400x150")
        self.grid_columnconfigure((0, 1), weight=1)

        self.button = customtkinter.CTkButton(self, text="my button", command=self.button_callback)
        self.button.grid(large_widget_sty)
        self.checkbox_1 = customtkinter.CTkCheckBox(self, text="checkbox 1")
        self.checkbox_1.grid(same_row_sty)
        self.checkbox_2 = customtkinter.CTkCheckBox(self, text="checkbox 2")
        self.checkbox_2.grid(same_row_sty)
        
    def button_callback(self):
        print("button pressed")

app = App()
app.mainloop()
```

**Comment by Akascape** [comment, marked as answer]
@Heus-Sueh Possible but needs manual work. You can make a dictionary and save the params one by one. Something like this program has: https://github.com/Akascape/CTkDesigner-Support where it saves all created widgets in json as a template file for reuse.

**Comment by Akascape** [comment]
@Heus-Sueh I also found this: https://github.com/PaulleDemon/tkStyleSheet , you can modify this package for ctk widgets.

**Comment by Heus-Sueh** [comment]
Thanks for the answer

[Off-Topic]
@Akascape why do you like CustomTkinter?

I always see you helping other people with this project, for me tkinter is old and the custom tkinter is limited to this old project that is tkinter.

What makes you continue working with him?

**Reply by Akascape** [reply]
@Heus-Sueh CustomTkinter is actually a great derivative of tkinter. I feel most of the ui elements are there in tkinter and customtkinter which is enoug to make an awesome modern UI. 
I like UI developement a lot. And I help people so that they can keep their work with customtkinter. Currently I believe, for python the best library for Ui development specially for beginner is customtkinter. Easy to use, and have all the modern features.

**Reply by Heus-Sueh** [reply]
Have you used other GUI toolkits? QT or GTK?

**Reply by Akascape** [reply]
I have tried other libraries like QT and they are also great. Everybody has their own taste. I like customtkinter when it comes to python ui development.

**Reply by Heus-Sueh** [reply]
I think QT is very powerful and customizable, but its layout system is a little complex, especially for beginners

**Reply by Heus-Sueh** [reply]
Have you tested this framework [Flet](https://github.com/flet-dev/flet)? looks very promising, but for now it only has widgets for mobile
